src/NServiceBus.Core.Tests/Config/When_users_override_the_configuration_source.cs

```diff
@@ -1,34 +1,55 @@
 namespace NServiceBus.Core.Tests.Config
 {
+    using System.Threading.Tasks;
     using NServiceBus.Config.ConfigurationSource;
+    using NServiceBus.Features;
+    using NServiceBus.Settings;
     using NUnit.Framework;
 
     [TestFixture]
     public class When_users_override_the_configuration_source
     {
-        IConfigurationSource userConfigurationSource;
-
-        Configure config;
-
-        [SetUp]
-        public void SetUp()
+        [Test]
+        public async Task NService_bus_should_resolve_configuration_from_that_source()
         {
-            userConfigurationSource = new UserConfigurationSource();
-
             var builder = new BusConfiguration();
-            builder.CustomConfigurationSource(userConfigurationSource);
 
-            config = builder.BuildConfiguration();
+            builder.SendOnly();
+            builder.TypesToScanInternal(new[] { typeof(ConfigSectionValidatorFeature) });
+            builder.EnableFeature<ConfigSectionValidatorFeature>();
+            builder.CustomConfigurationSource(new UserConfigurationSource());
+
+            var endpoint = await Endpoint.CreateAndStartAsync(builder);
+            await endpoint.StopAsync();
         }
-      
-        [Test]
-        public void NService_bus_should_resolve_configuration_from_that_source()
+
+        class ConfigSectionValidatorFeature : Feature
         {
-            var section = config.Settings.GetConfigSection<TestConfigurationSection>();
+            public ConfigSectionValidatorFeature()
+            {
+                RegisterStartupTask<ValidatorTask>();
+            }
 
-            Assert.AreEqual(section.TestSetting,"TestValue");
-        }
+            protected internal override void Setup(FeatureConfigurationContext context)
+            {
+            }
+
+            class ValidatorTask : FeatureStartupTask
+            {
+                ReadOnlySettings settings;
 
+                public ValidatorTask(ReadOnlySettings settings)
+                {
+                    this.settings = settings;
+                }
+
+                protected override void OnStart(ISendOnlyBus bus)
+                {
+                    var section = settings.GetConfigSection<TestConfigurationSection>();
+                    Assert.AreEqual(section.TestSetting, "TestValue");
```

Comment:
is there a way to move the assertion to the test method? This somehow looks more like an acceptance tests than an unit test?